src/helpers/client/FormValue.js

```diff
@@ -0,0 +1,109 @@
+/** Handles the core of form values, including:
+ * - Storing both a current and future value
+ * - Tracking when the user manipulates a future value
+ * - Propagating current values to future values if they
+ * haven't been manipulated by the user
+ * - Making this instance serializable
+ * - Making sure only valid values are saved.
+ * 
+ * All values stored in a FormValue instance should be valid
+ * 
+ * There must be a better way to write that.
+ * 
+ * @class
+ * @param {string} name - the name of the client property
+ * @param {any} [val] - the initial value
+ * @param {function} [makeValid] - returns a valid value no
+ * matter what. Will be given the new value, the current value,
+ * the string 'current' or 'future', and the `FormValue` instance.
+ * 
+ * @returns FormValue
+ * @todo Finish description of FormValue return value
+ */
+class FormValue {
+
+  constructor ( name, val = null, makeValid ) {
```

Comment:
If we're going to make one default explicit in the constructor, I would strongly suggest we make all defaults explicit in the constructor: IE if `val=null`, then we should have `makeValid = (newVal) => newVal` rather than mix and match. Otherwise, the next person to read the code may well expect that makeValid has no default because they expect the defaults to be explicit based on `val`.

Reply:
Fair point. I'd rather put the defaults in the body of the function then, so I'll do that.

Reply:
Fair enough.